### DISCUSSION TITLE
Utilize test containers to run e2e tests on various spark versions instead or relying on a locally installed one

### DISCUSSION THREAD

**Discussion by grazy27** [dotnet/spark#1195] [Ideas]
It's hard to test changes on old sparks locally

**Comment by wudanzy** [comment]
I planned to remove spark 2.X support after you PR.

**Comment by wudanzy** [comment]
2.x and even 3.0 is too old.

**Reply by grazy27** [reply]
Nice, that'd simplify things a lot. Test containers would still be handy for testing on linux, no need to manually configure 3.2 \ 3.3 \ 3.5 \ future versions, instead just download them pre-configured from docker hub. 

And for windows, we can either generate own containers or stick to existing approach on nightly
